### PR TITLE
Sympy-1.1 update

### PR DESCRIPTION
**Pull request by mlange05**
This merge adds the necessary compatibility changes to make Devito work with the new `sympy-1.1` and updates the dependencies everywhere.

**Note: This is not backwards compatible with `sympy-1.0` and all environments have to be upgraded!**

### CODE REVIEW

File: devito/dse/extended_sympy.py [hunk 1]
```diff
@@ -41,7 +41,16 @@ def xreplace(self, rule):
...
-    pass
+
+    """
+    A customized version of :class:`sympy.Eq` which suppresses
+    evaluation.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        kwargs['evaluate'] = False
+        obj = sympy.Eq.__new__(cls, *args, **kwargs)
```
Comment (FabioLuporini) [inline devito/dse/extended_sympy.py]: Just of out curiosity, where did it break w/o this ?

**Comment by mloubout** [comment]
Should `Eq` be imported from devito instead of `sympy` too in:
`examples/seismic/acoustic/operators.py`
`examples/seismic/tti/operators.py`
`devito/pointdata.py`

And any other one I may have missed?

File: devito/dse/extended_sympy.py (hunk 1 above)
Comment (mlange05) [reply devito/dse/extended_sympy.py]: The reason is subtle and only applies for a few corner-cases in `test_operator.py`; effectively degenerate expressions like `Eq(a, a + 3)` will now immediately evaluate to `False`, so we need to suppress evaluation.

**Comment by mlange05** [comment]
@mloubout Good call, I have updated as many as I could find. It technically doesn't matter, as only some degenerate expressions are affected (see comment above), but it's good for consistency.

File: devito/dse/extended_sympy.py (hunk 1 above)
Comment (mloubout) [reply devito/dse/extended_sympy.py]: Add `Eq` to `_all_` and should be good

**Comment by FabioLuporini** [comment]
Oh, perhaps time to replace ``q_indexed`` (in ``dse/queries``) with ``.is_Indexed`` all over the codebase ?

**Comment by FabioLuporini** [comment]
I think this should be the next one to go

**Comment by mlange05** [comment]
@FabioLuporini Agreed. I just rebased and tried to get rid of most of the `q_indexed` usages, although we still need it for the search engine. Should be good to go once travis is happy.

**Comment by FabioLuporini** [comment]
Thanks. Let's merge this as soon as Travis is happy

**Comment by FabioLuporini** [comment]
OK, merging...

**Comment by FabioLuporini** [comment]
Merged.

**Pull request merged**